### ISSUE TITLE
Missing highlighting for severity= hint

### ISSUE THREAD

**Issue by Narretz** [sublimelsp/LSP#891]
I have a situation where I only get highlighting for severity = error.
To illustrate, in typescript, the red outlined diagnostic is an error, and the one above is a hint (unused variable).
<img width="146" alt="2020-02-25 004210" src="https://user-images.githubusercontent.com/1153097/75201157-fb71ab00-5767-11ea-9a09-d30e50824f93.png">

But I can't figure out why I only get a red outline for the error severity.
The relevant config entries:
```
{
	"auto_show_diagnostics_panel": false,
	"show_diagnostics_severity_level": 4,
	"diagnostics_highlight_style": "box",
}
```

The hint shows up in the diagnostic panel, so I know it's there

**Comment by Narretz** [comment]
Oh I think I found the problem. In the following line
https://github.com/sublimelsp/LSP/blob/6a0afdfa99965326d12d26d06a72b54c09962c1d/plugin/diagnostics.py#L249

the call excludes `DiagnosticSeverity.Hint` because range() max is not inclusive. Essentially it stops at severity 3. Does that make sense?

**Comment by rwols** [comment]
I think this was intentional. The idea was, from memory, that the user doesn't want to get flooded with squigly linter lines that signify hints. See: https://github.com/sublimelsp/LSP/pull/800

**Comment by Narretz** [comment]
Oh, I see. I disagree with the decision to take this out of the user's hand completely. There should be an option that controls the severity highlight visibility, too. We can't just assume that no developer ever wants to see hint highlights.
Would a PR with an option be considered?

**Comment by rwols** [comment]
> Would a PR with an option be considered?

Sounds good to me!

I'm starting to think it *is* actually an off-by-one mistake due to the wording in this commit: https://github.com/sublimelsp/LSP/pull/800/commits/3c267eab898cf43778a1d05e6b582045b8812b4b

**Comment by Narretz** [comment]
Yeah "all regions" sounds like it's a bug.

**Comment by rchl** [comment]
Resolved by #941
